handlers/aws/utils.py

```diff
@@ -174,6 +174,7 @@ def get_shipper_from_input(
         anonymized_arn = anonymize_arn(event_input.id)
         input_has_output_type_telemetry(
             input_id=anonymized_arn.id,
+            input_type=event_input.type,
```

Comment:
`get_shipper_from_input` is called event by event for a single forwarder execution

we have three cases:
1. the original trigger is either `kinesis-data-stream`, `s3-sqs` or `sqs` (but not the continuing queue). in this case all the calls are idempotent
2. the original trigger is a `cloudwatch-logs`. in this case if the ARN of the input is that of a log group, all the calls are idempotent. at the contrary if the ARN is that of a log stream, the calls are not idempotent
3. the original trigger is the specific case of the `sqs` continuing queue. in this case the `event_input` object is indeed the one referencing the continued original trigger. in the same execution it can be any (`cloudwatch-logs`, `kinesis-data-stream`, `s3-sqs` or `sqs`). the calls are not idempotent

there are two things to consider here:
at https://github.com/elastic/elastic-serverless-forwarder/pull/156/commits/ee0b5a88bf8a63b4fd706121b3156d4c5059fcb4#diff-6fde71d82d3ead64aa933407ae918d5d91fbb4be69e09f236c2d5579c2cdb733R349 we send telemetry data every time `input_has_output_type_telemetry` is invoked: this happens multiple times. with that instead of sending a single telemetry call we send has much as the size of the lambda event.
moreover every single telemetry call will have the same payload where the data relative to the output gets growing, according if we are in one of the non idempotent cases

> the telemetry worker can send the data after successfully parsing the configuration while the main thread is working on the business data.

I'm fine with this approach, but then we should have
```
if isinstance(event, FunctionStartedEvent):
```

at https://github.com/elastic/elastic-serverless-forwarder/pull/156/commits/ee0b5a88bf8a63b4fd706121b3156d4c5059fcb4#diff-6fde71d82d3ead64aa933407ae918d5d91fbb4be69e09f236c2d5579c2cdb733R349

changing to `InputHasOutputTypeEvent` drastically increases the number of times we send data to the telemetry server :)

Reply:
Yeah, noticed the for loop a few lines above right after I pushed the changes ^^

I was considering sending the telemetry data after the config is successfully parsed.